service/core/kafka/server/src/main/scala/com/lightbend/lagom/internal/broker/kafka/Producer.scala

```diff
@@ -50,22 +53,14 @@ private[lagom] object Producer {
       offsetStore
     )
 
-    val backoffPublisherProps = BackoffOpts
-      .onStop(
-        publisherProps,
-        s"producer",
-        producerConfig.minBackoff,
-        producerConfig.maxBackoff,
-        producerConfig.randomBackoffFactor
-      )
-      .withDefaultStoppingStrategy
+    val entityIds = tags.toSet
 
     val clusterShardingSettings = ClusterShardingSettings(system).withRole(producerConfig.role)
 
     ClusterDistribution(system).start(
-      s"kafkaProducer-$topicId",
-      BackoffSupervisor.props(backoffPublisherProps),
-      tags.toSet,
+      projectionName,
+      topicProducerProps,
+      entityIds,
```

Comment:
done some cleanup to make `Producer` look even more similar to `ReadSideImpl`.

`Producer.scala` is to `ReadSideImpl` what `TopicProducerActor` is to `ReadSideActor`.